features/admin.roles.v2/components/edit-role/edit-role.tsx

```diff
@@ -127,7 +129,8 @@ export const EditRole: FunctionComponent<EditRoleProps> = (props: EditRoleProps)
                 render: () => (
                     <ResourceTab.Pane controlledSegmentation attached={ false }>
                         <BasicRoleDetails
-                            isReadOnly={ isAdminRole || isEveryoneRole || isReadOnly || isSharedRole }
+                            isReadOnly={ isAdminRole || isEveryoneRole || isReadOnly || isSharedRole
+                                || roleObject?.displayName === accountAppImpersonateRoleName }
```

Comment:
instead of comparing the display name, shall we use `role.meta.systemRole` property? (I'm assuming the myaccount impersonation role is a system role. correct me if im wrong)